src/adapters/adkernel.js

```diff
@@ -0,0 +1,194 @@
+     * @param imps collection of impressions
+     */
+    function buildRtbRequest(imps) {
+      return {
+        'id': utils.getUniqueIdentifierStr(),
+        'imp': imps,
+        'site': site,
+        'at': 1,
+        'device': {
+          'ip': 'caller',
+          'ua': 'caller'
+        }
+      };
+    }
+
+    /**
+     * Creates site description object
+     */
+    function createSite() {
+      var location = utils.getTopWindowLocation();
+      return {
+        'domain': location.hostname,
+        'page': location.pathname
+      };
+    }
+  }
+
+  /**
+   *  Main module export function implementation
+   */
+  function _callBids(params) {
+    var bids = params.bids || [];
+    processBids(bids);
+  }
+
+  /**
+   *  Process all bids grouped by network/zone
+   */
+  function processBids(bids) {
+    const dispatcher = new RtbRequestDispatcher();
+    //process individual bids
+    utils._each(bids, function (bid) {
+      let size = bid.sizes[0];
+      dispatcher.addImp(bid.params.host, bid.params.zoneId, size[0], size[1], bid.bidId, bid.placementCode);
+    });
+    //process bids grouped into bidrequests
+    dispatcher.dispatch(function (host, zone, rtbRequest, imp2plcmnt) {
+      let url = buildEndpointUrl(host);
+      let params = buildRequestParams(zone, rtbRequest);
+      let bidHandlerCallback = createBidResponseHandler(rtbRequest, imp2plcmnt);
+      ajax.ajax(url, bidHandlerCallback, params, {
+        contentType: 'text/plain',
+        withCredentials: true,
+        method: 'GET'
+      });
+    });
+  }
+
+  /**
+   *  Creates callback function to process bid response
+   */
+  function createBidResponseHandler(rtbRequest, imp2adUnit) {
+    return function (bidResp) {
+      bidResp = bidResp === '' ? emptyBidResponse() : JSON.parse(bidResp);
+      utils._each(rtbRequest.imp, function (entry) {
+        let adUnitId = imp2adUnit[entry.id],
+          bidFound = false;
+        utils._each(bidResp.seatbid[0].bid, function (bid) {
+          if (!bidFound && bid.impid === entry.id) {
+            bidFound = true;
+            utils.logMessage('got response for ' + adUnitId);
+            bidmanager.addBidResponse(adUnitId, createBidObject(bid, entry.banner.w, entry.banner.h));
+          }
+        });
+        if (!bidFound) {
+          utils.logMessage('got empty response for ' + adUnitId);
+          bidmanager.addBidResponse(adUnitId, createEmptyBidObject());
+        }
+      });
+    };
+  }
+
+  /**
+   * Build ad-network specific endpoint url
+   */
+  function buildEndpointUrl(host) {
+    return window.location.protocol + '//' + host + '/rtbg';
+  }
+
+  function buildRequestParams(zone, rtbReq) {
+    return {
+      'zone': zone,
+      'ad_type': 'rtb',
+      'r': encodeURIComponent(JSON.stringify(rtbReq))
+    };
+  }
+
+  /**
+   *  Create bid object for the bid manager
+   */
+  function createBidObject(bidObj, width, height) {
+    return utils.extend(bidfactory.createBid(1), {
```

Comment:
Please store a reference to the bid request object and pass it to `createBid(1, bidRequest)` as this allows us to match the bid.adId to bid response.